File: src/main/java/io/github/kingg22/api/vacunas/panama/service/IPdfService.java
package io.github.kingg22.api.vacunas.panama.service;

import io.github.kingg22.api.vacunas.panama.web.dto.DosisDto;
import io.github.kingg22.api.vacunas.panama.web.dto.PacienteDto;
import java.io.IOException;
import java.util.List;
import java.util.UUID;

public interface IPdfService {
    byte[] generatePdf(PacienteDto pacienteDto, List<DosisDto> dosisDtos, UUID idCertificate) throws IOException;

    String generatePdfBase64(PacienteDto pacienteDto, List<DosisDto> dosisDtos, UUID idCertificate) throws IOException;
}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/service/IPdfService.kt
package io.github.kingg22.api.vacunas.panama.service

import io.github.kingg22.api.vacunas.panama.web.dto.DosisDto
import io.github.kingg22.api.vacunas.panama.web.dto.PacienteDto
import java.io.IOException
import java.util.UUID

interface IPdfService {
    @Throws(IOException::class)
    fun generatePdf(pacienteDto: PacienteDto, dosisDtos: List<DosisDto>, idCertificate: UUID): ByteArray

    @Throws(IOException::class)
    fun generatePdfBase64(pacienteDto: PacienteDto, dosisDtos: List<DosisDto>, idCertificate: UUID): String
}
